message_definitions/v1.0/common.xml

```diff
@@ -5711,21 +5711,21 @@
       <field type="uint32_t" name="time_boot_ms" units="ms">Timestamp (time since system boot).</field>
       <field type="uint8_t[32]" name="vendor_name">Name of the camera vendor</field>
       <field type="uint8_t[32]" name="model_name">Name of the camera model</field>
-      <field type="uint32_t" name="firmware_version">Version of the camera firmware, encoded as: (Dev &amp; 0xff) &lt;&lt; 24 | (Patch &amp; 0xff) &lt;&lt; 16 | (Minor &amp; 0xff) &lt;&lt; 8 | (Major &amp; 0xff)</field>
-      <field type="float" name="focal_length" units="mm">Focal length</field>
-      <field type="float" name="sensor_size_h" units="mm">Image sensor size horizontal</field>
-      <field type="float" name="sensor_size_v" units="mm">Image sensor size vertical</field>
-      <field type="uint16_t" name="resolution_h" units="pix">Horizontal image resolution</field>
-      <field type="uint16_t" name="resolution_v" units="pix">Vertical image resolution</field>
-      <field type="uint8_t" name="lens_id">Reserved for a lens ID</field>
+      <field type="uint32_t" name="firmware_version">Version of the camera firmware, encoded as: (Dev &amp; 0xff) &lt;&lt; 24 | (Patch &amp; 0xff) &lt;&lt; 16 | (Minor &amp; 0xff) &lt;&lt; 8 | (Major &amp; 0xff).  Use UINT32_MAX if not known.</field>
+      <field type="float" name="focal_length" units="mm">Focal length.  Use NaN is not known.</field>
+      <field type="float" name="sensor_size_h" units="mm">Image sensor size horizontal.  Use NaN is not known.</field>
+      <field type="float" name="sensor_size_v" units="mm">Image sensor size vertical.  Use NaN is not known.</field>
+      <field type="uint16_t" name="resolution_h" units="pix">Horizontal image resolution.  Use 0 if not known.</field>
+      <field type="uint16_t" name="resolution_v" units="pix">Vertical image resolution.  Use 0 if not known.</field>
+      <field type="uint8_t" name="lens_id">Reserved for a lens ID.  Use UINT8_MAX if not known.</field>
       <field type="uint32_t" name="flags" enum="CAMERA_CAP_FLAGS" display="bitmask">Bitmap of camera capability flags.</field>
-      <field type="uint16_t" name="cam_definition_version">Camera definition version (iteration)</field>
-      <field type="char[140]" name="cam_definition_uri">Camera definition URI (if any, otherwise only basic functions will be available). HTTP- (http://) and MAVLink FTP- (mavlinkftp://) formatted URIs are allowed (and both must be supported by any GCS that implements the Camera Protocol).</field>
+      <field type="uint16_t" name="cam_definition_version">Camera definition version (iteration).  Use 0 if not known.</field>
+      <field type="char[140]" name="cam_definition_uri">Camera definition URI (if any, otherwise only basic functions will be available). HTTP- (http://) and MAVLink FTP- (mavlinkftp://) formatted URIs are allowed (and both must be supported by any GCS that implements the Camera Protocol).  Use a zero-length string if not known.</field>
```

Comment:
Yes. Is this the normal way to describe a zero length string? i.e. don't we usually use something like "null terminated"?

Reply:
zero-length is the value; the encoding of a string in mavlink is "null terminated or the length of the field", so I think this is fine :-)